scripts/ecs-anywhere-install.sh

```diff
@@ -0,0 +1,526 @@
+    apt)
+        curl -o "$dir/$DEB_PKG_NAME" "$DEB_URL"
+        if $CHECK_SHA; then
+            curl -o "$dir/$DEB_PKG_NAME.asc" "$DEB_URL.asc"
+            ecs-init-signature-verify "$dir/$DEB_PKG_NAME.asc" "$dir/$DEB_PKG_NAME"
+        fi
+        apt install -y "$dir/$DEB_PKG_NAME"
+        rm -rf "$dir"
+        ;;
+    dnf | yum | zypper)
+        curl -o "$dir/$RPM_PKG_NAME" "$RPM_URL"
+        if $CHECK_SHA; then
+            curl -o "$dir/$RPM_PKG_NAME.asc" "$RPM_URL.asc"
+            ecs-init-signature-verify "$dir/$RPM_PKG_NAME.asc" "$dir/$RPM_PKG_NAME"
+        fi
+        local args="-y"
+        if [[ "$PKG_MANAGER" == "zypper" ]]; then
+            args="${args} --allow-unsigned-rpm"
+        fi
+        $PKG_MANAGER install ${args} "$dir/$RPM_PKG_NAME"
+        rm -rf "$dir"
+        ;;
+    esac
+
+    if [ ! -f "/etc/ecs/ecs.config" ]; then
+        touch /etc/ecs/ecs.config
+    else
+        echo "/etc/ecs/ecs.config already exists, preserving existing config and appending cluster name."
+    fi
+    echo "ECS_CLUSTER=$ECS_CLUSTER" >>/etc/ecs/ecs.config
+
+    if [ ! -f "/var/lib/ecs/ecs.config" ]; then
+        touch /var/lib/ecs/ecs.config
+    else
+        echo "/var/lib/ecs/ecs.config already exists, preserving existing config and appending ECS anywhere requirements."
+    fi
+    echo "AWS_DEFAULT_REGION=$REGION" >>/var/lib/ecs/ecs.config
+    echo "ECS_EXTERNAL=true" >>/var/lib/ecs/ecs.config
+    if [ ! -z "$ECS_ENDPOINT" ]; then
+        echo "ECS_BACKEND_HOST=$ECS_ENDPOINT" >>/var/lib/ecs/ecs.config
+    fi
+    systemctl enable ecs
+    systemctl start ecs
+
+    ok
+}
+
+ecs-init-signature-verify() {
+    echo "Verifying the signature of ECS Init Artifact"
+
+    #TODO [Update before release] Update links here to use prod urls (or urls specified by $DEB_URL or $RPM_URL)
+    curl -o "$dir/amazon-ecs-init.gpg" "https://ecs-init-packages-testing.s3.amazonaws.com/amazon-ecs-public-key.gpg"
+    gpg --import "$dir/amazon-ecs-init.gpg"
+
+    gpg --verify "$1" "$2" &> "$dir/verifyOut"
+
+    if grep -Fxq "BAD signature" "$dir/verifyOut"; then
+        echo "ECS Init GPG verification fail. Stop the installation of the ECS Agent. Please contact AWS Support."
+        fail
+    else
+        echo "ECS Init GPG verification passed. Install the ECS Agent."
+    fi
+    rm "$dir/verifyOut"
+
+    ok
+}
+
+verify-agent() {
+    retryLimit=10
+    i=0
+    success=0
+    for ((i = 0 ; i < retryLimit ; i++))
+    do
+        curlResult="$(timeout 10 curl -s http://localhost:51678/v1/metadata | jq .ContainerInstanceArn)"
+        echo "container instance arn: $curlResult"
+        if [ ! "$curlResult" == "null" ] && [ -n "$curlResult" ]; then
+            success=1
+            break
+        fi
+    done
+    if [ $success == 0 ]; then
+        echo "ECS Agent verification failed. Please check logs at /var/log/ecs/ecs-agent.log"
+        fail
+    else
+        echo "ping ECS Agent registered successfully!"
```

Comment:
nit: since we got the container instance arn, maybe just print it here for people to see for convenience

Reply:
Sure thing.

Reply:
I actually have it at above lines. I will combine the two.